From e7ae70d261fcf002f9db1c2214252f4a5f53a98b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Wed, 22 May 2024 15:15:09 +0200
Subject: [PATCH 604/841] feat(stm32mp21): add HASH node for STM32MP21
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add the hash node to STM32MP21 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 7dff19a9cc7c3e3761c8e1a669f56d9438ec392b feat(stm32mp21): add HASH node for STM32MP21]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: I3d0d593cf4bf82452c484cba0955a4505b6a2eea
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/381684
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 fdts/stm32mp211.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/fdts/stm32mp211.dtsi b/fdts/stm32mp211.dtsi
index 62029553a..d66fec410 100644
--- a/fdts/stm32mp211.dtsi
+++ b/fdts/stm32mp211.dtsi
@@ -197,6 +197,14 @@
 				status = "disabled";
 			};
 
+			hash1: hash@42030400 {
+				compatible = "st,stm32mp13-hash";
+				reg = <0x42030400  0x400>;
+				clocks = <&rcc CK_BUS_HASH1>;
+				resets = <&rcc HASH1_R>;
+				status = "disabled";
+			};
+
 			rng2: rng@42020000 {
 				compatible = "st,stm32mp13-rng";
 				reg = <0x42020000 0x400>;
-- 
2.47.2

